### PR TITLE
add reincrementalify to completionGoals 

### PR DESCRIPTION
**Pull request by casz**
cc @jglick @oleg-nenashev 

This could be applied to https://github.com/jenkinsci/pom as well 👍 

Here we can see the completionGoals in use: jenkinsci/kotlin-v1-stdlib-jdk8-plugin@afc6746

### CODE REVIEW

**Comment by ndeloof** [comment]
will this extra goal be a no-op on plugins without increment extension ?

**Comment by casz** [comment]
Oh I can do one better 😃 I can just check for "might-produce-incrementals" property!

**Comment by oleg-nenashev** [comment]
might != will though

**Comment by casz** [comment]
No but if the profile is activate it means when you release you still get the benefit of completetion goals sadly maven is limited once again and you cannot chain profiles

**Comment by casz** [comment]
I can't really see a good way of activating this profile without assuming you are using incremental if you have `.mvn/config` if we make reincrementalify check if revision and changelist are there it before changing the pom we could be made fault tolerant and not break anybody.

**Comment by jglick** [comment]
Yes something similar should go into `jenkinsci/pom` but beware that @kohsuke’s automated script for releasing `jenkins` already reincrementalify and commit and push, so it would probably break—we would need to coördinate with him to turn that off.

**Comment by casz** [comment]
Followed your advice @jglick 👍

**Comment by jglick** [comment]
[JENKINS-51658](https://issues.jenkins-ci.org/browse/JENKINS-51658) FTR

File: pom.xml [hunk 1]
```diff
@@ -1305,6 +1305,9 @@
           </plugin>
         </plugins>
       </build>
+      <properties>
```
Comment (jglick) [inline pom.xml]: As noted in https://github.com/jenkinsci/pom/pull/27#pullrequestreview-137653976 this would better be done using mojo configuration.

**Comment by casz** [comment]
I'll do a follow up :blush:

**Pull request merged**